src/coord/coord.rs

```diff
@@ -1650,8 +1650,56 @@ where
         self.import_source_or_view(id, &index.on, &mut dataflow);
         dataflow.add_index_to_build(*id, index.on.clone(), on_type.clone(), index.keys.clone());
         dataflow.add_index_export(*id, index.on, on_type, index.keys.clone());
-        self.broadcast_dataflow_creation(dataflow);
         self.insert_index(*id, &index, self.logical_compaction_window_ms);
+        self.validate_dataflow(&mut dataflow);
+        self.broadcast_dataflow_creation(dataflow);
+    }
+
+    /// Performs some sanity checking on the dataflow before shipping it.
+    ///
+    /// In particular, there are requirement on the `as_of` field for the dataflow
+    /// and the `since` frontiers of created arrangements, as a function of the `since`
+    /// frontiers of dataflow inputs (sources and imported arrangements).
+    fn validate_dataflow(&mut self, dataflow: &mut DataflowDesc) {
```

Comment:
This all looks right to me - I want to sanity check my understanding that `since` can be in advance of `upper` right? That was what I got from reading the comments at the top of `description.rs` and that's why we don't need to ensure that `since` is less than any of the `upper`'s in this dataflow?

Reply:
If we always want to call this function before `broadcast_dataflow_creation` maybe it's slightly better to place the function call inside of the `broadcast_dataflow_creation` body?

Reply:
Since can be in advance of `upper`, yes.

Yeah, I'm not sure how to organize this. We would like to make sure that the validate logic gets called, but messing around with `since` fields seems like it doesn't fall under the name of `broadcast_*`. Perhaps something else that looks like `finalize_dataflow` and consumes the description.

Reply:
it looks like we do optimization inside of `broadcast` too which I guess we could throw in `finalize` as well if we want to reorganize. regardless, this looks fine to merge!